packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -3265,38 +3273,54 @@ export default class Meeting extends StatelessWebexPlugin {
 
           if (this.config.stats.enableStatsAnalyzer) {
             // TODO: ** Dont re create StatsAnalyzer on reconnect or rejoin
-            this.statsAnalyzer = new StatsAnalyzer(this.config.stats);
+            this.networkQualityMonitor = new NetworkQualityMonitor(this.config.stats);
+            this.statsAnalyzer = new StatsAnalyzer(this.config.stats, this.networkQualityMonitor);
             this.statsAnalyzer.on(EVENT_TRIGGERS.MEDIA_QUALITY, (options) => {
               // TODO:  might have to send the same event to the developer
               // Add ip address info if geo hint is present
               options.data.intervalMetadata.peerReflexiveIP = this.webex.meetings.geoHintInfo?.clientAddress || options.data.intervalMetadata.peerReflexiveIP || MQA_STATS.DEFAULT_IP;
               Metrics.postEvent({event: eventType.MEDIA_QUALITY, meeting: this, data: {intervalData: options.data, networkType: options.networkType}});
             });
-            this.statsAnalyzer.on(EVENT_TRIGGERS.MEETING_HIGH_PACKETLOSS, (res) => {
-              if (res.mediaType === VIDEO) {
-                if (this.config.stats.autoDowngradeEnabled && this.canUpdateMedia()) {
-                  LoggerProxy.logger.log(`${LOG_HEADER} High Packet Loss Detected, Adjusting Local Video Quality`);
-                  if (this.mediaProperties.localQualityLevel === QUALITY_LEVELS.HIGH) {
-                    this.setLocalVideoQuality(QUALITY_LEVELS.MEDIUM);
-                  }
-                  else if (this.mediaProperties.localQualityLevel === QUALITY_LEVELS.MEDIUM) {
-                    this.setLocalVideoQuality(QUALITY_LEVELS.LOW);
-                  }
+            this.networkQualityMonitor.on(EVENT_TRIGGERS.NETWORK_QUALITY_JUDGMENT, (res) => {
+              Trigger.trigger(
+                this,
+                {
+                  file: 'meeting/index',
+                  function: 'addMedia'
+                },
+                EVENT_TRIGGERS.NETWORK_QUALITY_JUDGMENT,
+                {
+                  networkAcceptable: res.networkAcceptable,
+                  mediaType: res.mediaType,
+                  data: res.data
                 }
-                Trigger.trigger(
-                  this,
-                  {
-                    file: 'meeting/index',
-                    function: 'addMedia'
-                  },
-                  EVENT_TRIGGERS.MEETING_HIGH_PACKETLOSS,
-                  {
-                    mediaType: res.mediaType,
-                    packetLossRatio: res.packetLossRatio
-                  }
-                );
-              }
+              );
             });
+            // this.statsAnalyzer.on(EVENT_TRIGGERS.MEETING_HIGH_PACKETLOSS, (res) => {
```

Comment:
What to do with current HIGH_PACKETLOSS - event do we keep and tell them event has been deprecated or remove all together would be a breaking change

Reply:
its fine to remove it and customer to use the network quality

Reply:
right my concern was if someone is using it properly and udpates the SDK it would break no? How do we alert them to use the new event?